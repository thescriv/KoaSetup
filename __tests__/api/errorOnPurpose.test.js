const { startApi, stopApi } = require("../../src/api")
const { createApiClient } = require("../utils")

const config = require("../../src/config")

describe("errorOnPurpose API", () => {
  let apiClient = null

  beforeAll(async () => {
    await startApi(3001)

    const apiUrl = `${config.API_URL}:3001`

    apiClient = createApiClient(apiUrl)
  })

  afterAll(async () => {
    await stopApi()
  })

  describe("errorOnPurposeAPI", () => {
    test("do not fetch errorOnPurpose", async () => {
      let error

      try {
        await apiClient.errorOnPurpose()
      } catch (err) {
        error = err.response
      }
      const { body, status } = error

      expect({ body, status }).toMatchSnapshot()
    })
  })
})
