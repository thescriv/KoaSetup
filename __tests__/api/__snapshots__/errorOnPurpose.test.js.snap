// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorOnPurpose API errorOnPurposeAPI do not fetch errorOnPurpose 1`] = `
Object {
  "body": Object {
    "message": "An error occured, sorry !",
  },
  "status": 500,
}
`;
